src/datastore/postgres-store.ts

```diff
@@ -5476,22 +5476,21 @@ export class PgDataStore
         `
         SELECT name FROM (
           (
-            SELECT DISTINCT ON (name) name, registered_at as block_height, tx_index
+            SELECT DISTINCT ON (name) name, registered_at as block_height, tx_index, address
             FROM names
-            WHERE address = $1
-            AND registered_at <= $2
+            WHERE registered_at <= $2
             AND canonical = true AND microblock_canonical = true
             ORDER BY name, registered_at DESC, tx_index DESC
           )
           UNION ALL (
-            SELECT DISTINCT ON (fully_qualified_subdomain) fully_qualified_subdomain as name, block_height, tx_index
+            SELECT DISTINCT ON (fully_qualified_subdomain) fully_qualified_subdomain as name, block_height, tx_index, owner as address
             FROM subdomains
-            WHERE owner = $1
```

Comment:
The only reason this query was performant enough is because of this index:
```ts
  pgm.createIndex('subdomains', [
    { name: 'fully_qualified_subdomain' },
    { name: 'canonical', sort: 'DESC' },
    { name: 'microblock_canonical', sort: 'DESC' },
    { name: 'block_height', sort: 'DESC' },
    { name: 'tx_index', sort: 'DESC' },
  ]);
```

Adding the extra `owner` column to the query makes the index unusable. 

I'm not sure we could just add `owner` to the index because it looks like this query would end up gathering _all_ the subdomains in memory before excluding entries in the outer WHERE clause. We'd need to run some manual benchmarks on this query against the fully imported subdomain list.

Reply:
@zone117x You were right about it, this approach was taking 129666ms(129 seconds). 
I have updated it, the new approach takes 137ms for the main query, and the second query will take 137* N (N=number of rows returned in the first query).